script/spec-runner.js

```diff
@@ -186,9 +186,9 @@ async function runMainProcessElectronTests () {
 }
 
 async function installSpecModules (dir) {
-  // v8 headers use c++17 so override the gyp default of -std=c++14,
+  // v8 headers use c++20 so override the gyp default of -std=c++14,
   // but don't clobber any other CXXFLAGS that were passed into spec-runner.js
-  const CXXFLAGS = ['-std=c++17', process.env.CXXFLAGS].filter(x => !!x).join(' ');
+  const CXXFLAGS = ['-std=c++20', process.env.CXXFLAGS].filter(x => !!x).join(' ');
```

Comment:
```suggestion
  const CXXFLAGS = ['-std=gnu++20', process.env.CXXFLAGS].filter(x => !!x).join(' ');
```

if the CI uses gcc <= 9 then it would be `-std=gnu++2a` but no need to make the condition if it passes with the above.

Reply:
Not new to this PR -- for example, I see we had `gnu++17` before -- but does anyone know if the GNU extensions are actually needed here?

Unless we need the GNU extensions, we'd be better off using the standard, i.e. `-std=c++20` here and in the patch above

Reply:
We don't use it but the headers from Node.js could, but I missed this before but we should actually remove the `-std` argument from this list, it will get set by `common.gypi` when building the native modules. We were overriding previously when headers had c++14 and we wanted to use c++17.

Reply:
@deepak1556 done in dc1e1bb